<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LibElasParametersControlWidget</class>
 <widget class="QWidget" name="LibElasParametersControlWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LibElasParameters</string>
  </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="margin">
     <number>3</number>
    </property>
    <property name="spacing">
     <number>3</number>
    </property>
    <item row="0" column="0">
     <widget class="QLabel" name="label_0">
       <property name="text">
         <string>Disp Min</string>
       </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QSpinBox" name="dispMinSpinBox">
       <property name="toolTip">
         <string>min disparity</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>2550</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_1">
       <property name="text">
         <string>Disp Max</string>
       </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QSpinBox" name="dispMaxSpinBox">
       <property name="toolTip">
         <string>max disparity</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>2550</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>255</number>
       </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_2">
       <property name="text">
         <string>Support Threshold</string>
       </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QDoubleSpinBox" name="supportThresholdSpinBox">
       <property name="toolTip">
         <string>maximum uniqueness ratio best vs second best support match</string>
       </property>
       <property name="minimum">
        <double>0</double>
       </property>
       <property name="maximum">
        <double>1</double>
       </property>
       <property name="singleStep">
        <double>0.05</double>
       </property>
       <property name="value">
        <double>0.85</double>
       </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_3">
       <property name="text">
         <string>Support Texture</string>
       </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QSpinBox" name="supportTextureSpinBox">
       <property name="toolTip">
         <string>min texture for support points</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_4">
       <property name="text">
         <string>Candidate Stepsize</string>
       </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QSpinBox" name="candidateStepsizeSpinBox">
       <property name="toolTip">
         <string>step size of regular grid on which support points are matched</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>5</number>
       </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_5">
       <property name="text">
         <string>Incon Window Size</string>
       </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QSpinBox" name="inconWindowSizeSpinBox">
       <property name="toolTip">
         <string>window size of inconsistent support point check</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>5</number>
       </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="label_6">
       <property name="text">
         <string>Incon Threshold</string>
       </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QSpinBox" name="inconThresholdSpinBox">
       <property name="toolTip">
         <string>disparity similarity threshold for support point to be considered consistent</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>5</number>
       </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="label_7">
       <property name="text">
         <string>Incon Min Support</string>
       </property>
     </widget>
    </item>
    <item row="7" column="1">
     <widget class="QSpinBox" name="inconMinSupportSpinBox">
       <property name="toolTip">
         <string>minimum number of consistent support points</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>5</number>
       </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QLabel" name="label_8">
       <property name="text">
         <string>Add Corners</string>
       </property>
     </widget>
    </item>
    <item row="8" column="1">
     <widget class="QCheckBox" name="addCornersCheckBox">
       <property name="toolTip">
         <string>add support points at image corners with nearest neighbor disparities</string>
       </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QLabel" name="label_9">
       <property name="text">
         <string>Grid Size</string>
       </property>
     </widget>
    </item>
    <item row="9" column="1">
     <widget class="QSpinBox" name="gridSizeSpinBox">
       <property name="toolTip">
         <string>size of neighborhood for additional support point extrapolation</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>20</number>
       </property>
     </widget>
    </item>
    <item row="10" column="0">
     <widget class="QLabel" name="label_10">
       <property name="text">
         <string>beta</string>
       </property>
     </widget>
    </item>
    <item row="10" column="1">
     <widget class="QDoubleSpinBox" name="betaSpinBox">
       <property name="toolTip">
         <string>image likelihood parameter</string>
       </property>
       <property name="minimum">
        <double>0</double>
       </property>
       <property name="maximum">
        <double>1</double>
       </property>
       <property name="singleStep">
        <double>0.05</double>
       </property>
       <property name="value">
        <double>0.02</double>
       </property>
     </widget>
    </item>
    <item row="11" column="0">
     <widget class="QLabel" name="label_11">
       <property name="text">
         <string>gamma</string>
       </property>
     </widget>
    </item>
    <item row="11" column="1">
     <widget class="QDoubleSpinBox" name="gammaSpinBox">
       <property name="toolTip">
         <string>prior constant</string>
       </property>
       <property name="minimum">
        <double>0</double>
       </property>
       <property name="maximum">
        <double>10</double>
       </property>
       <property name="singleStep">
        <double>1</double>
       </property>
       <property name="value">
        <double>3</double>
       </property>
     </widget>
    </item>
    <item row="12" column="0">
     <widget class="QLabel" name="label_12">
       <property name="text">
         <string>sigma</string>
       </property>
     </widget>
    </item>
    <item row="12" column="1">
     <widget class="QDoubleSpinBox" name="sigmaSpinBox">
       <property name="toolTip">
         <string>prior sigma</string>
       </property>
       <property name="minimum">
        <double>0</double>
       </property>
       <property name="maximum">
        <double>10</double>
       </property>
       <property name="singleStep">
        <double>1</double>
       </property>
       <property name="value">
        <double>1</double>
       </property>
     </widget>
    </item>
    <item row="13" column="0">
     <widget class="QLabel" name="label_13">
       <property name="text">
         <string>S Radius</string>
       </property>
     </widget>
    </item>
    <item row="13" column="1">
     <widget class="QDoubleSpinBox" name="sRadiusSpinBox">
       <property name="toolTip">
         <string>prior sigma radius</string>
       </property>
       <property name="minimum">
        <double>0</double>
       </property>
       <property name="maximum">
        <double>10</double>
       </property>
       <property name="singleStep">
        <double>1</double>
       </property>
       <property name="value">
        <double>2</double>
       </property>
     </widget>
    </item>
    <item row="14" column="0">
     <widget class="QLabel" name="label_14">
       <property name="text">
         <string>Match Texture</string>
       </property>
     </widget>
    </item>
    <item row="14" column="1">
     <widget class="QSpinBox" name="matchTextureSpinBox">
       <property name="toolTip">
         <string>min texture for dense matching</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
     </widget>
    </item>
    <item row="15" column="0">
     <widget class="QLabel" name="label_15">
       <property name="text">
         <string>L R Threshold</string>
       </property>
     </widget>
    </item>
    <item row="15" column="1">
     <widget class="QSpinBox" name="lRThresholdSpinBox">
       <property name="toolTip">
         <string>disparity threshold for left right consistency check</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
     </widget>
    </item>
    <item row="16" column="0">
     <widget class="QLabel" name="label_16">
       <property name="text">
         <string>Speckle Sim Threshold</string>
       </property>
     </widget>
    </item>
    <item row="16" column="1">
     <widget class="QDoubleSpinBox" name="speckleSimThresholdSpinBox">
       <property name="toolTip">
         <string>similarity threshold for speckle segmentation</string>
       </property>
       <property name="minimum">
        <double>0</double>
       </property>
       <property name="maximum">
        <double>10</double>
       </property>
       <property name="singleStep">
        <double>1</double>
       </property>
       <property name="value">
        <double>1</double>
       </property>
     </widget>
    </item>
    <item row="17" column="0">
     <widget class="QLabel" name="label_17">
       <property name="text">
         <string>Speckle Size</string>
       </property>
     </widget>
    </item>
    <item row="17" column="1">
     <widget class="QSpinBox" name="speckleSizeSpinBox">
       <property name="toolTip">
         <string>maximal size of a speckle. Small speckles get removed</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>200</number>
       </property>
     </widget>
    </item>
    <item row="18" column="0">
     <widget class="QLabel" name="label_18">
       <property name="text">
         <string>Ipol Gap Width</string>
       </property>
     </widget>
    </item>
    <item row="18" column="1">
     <widget class="QSpinBox" name="ipolGapWidthSpinBox">
       <property name="toolTip">
         <string>interpolate small gaps</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>3</number>
       </property>
     </widget>
    </item>
    <item row="19" column="0">
     <widget class="QLabel" name="label_19">
       <property name="text">
         <string>Filter Median</string>
       </property>
     </widget>
    </item>
    <item row="19" column="1">
     <widget class="QCheckBox" name="filterMedianCheckBox">
       <property name="toolTip">
         <string>optional median filter approximated</string>
       </property>
     </widget>
    </item>
    <item row="20" column="0">
     <widget class="QLabel" name="label_20">
       <property name="text">
         <string>Filter Adaptive Mean</string>
       </property>
     </widget>
    </item>
    <item row="20" column="1">
     <widget class="QCheckBox" name="filterAdaptiveMeanCheckBox">
       <property name="toolTip">
         <string>optional adaptive mean filter approximated</string>
       </property>
       <property name="checked">
           <bool>true</bool>
       </property>
     </widget>
    </item>
   <item row="21" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="21" column="2">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="3" margin="3"/>
 <resources/>
 <connections>
  <connection>
   <sender>dispMinSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>dispMaxSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>supportThresholdSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>supportTextureSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>candidateStepsizeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>inconWindowSizeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>inconThresholdSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>inconMinSupportSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>addCornersCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>gridSizeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>betaSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>gammaSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>sigmaSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>sRadiusSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>matchTextureSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>lRThresholdSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>speckleSimThresholdSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>speckleSizeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>ipolGapWidthSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>filterMedianCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>filterAdaptiveMeanCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>LibElasParametersControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
 </connections>
 <slots>
  <slot>changeParameters()</slot>
 </slots>
 <customwidgets>
 </customwidgets>
</ui>
